### PR TITLE
Terraform in Equinix Metal: Fix NAT to reference correct interfaces

### PR DESCRIPTION
**Pull request by nshalman**
## Description

The NAT setup commands assume that the interface is named eth1, when clearly from the [documentation](https://github.com/tinkerbell/tinkerbell-docs/blame/master/docs/setup/equinix-metal-terraform.md#L118) it is named `enp1s0f1`. This commit fixes the NAT setup commands accordingly.

## Why is this needed

NAT doesn't work by default on Equinix Metal when following the documentation

## How Has This Been Tested?

- [x] Tested with Terraform in Equinix Metal

## How are existing users impacted? What migration steps/scripts do we need?

Existing sandboxes (that are broken) should either be rebuilt, or can run the commands manually to enable NAT

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

**Comment by nshalman** [comment]
Still need to do a full reprovision test. The commands worked manually.

**Review by markyjackson-taulia — APPROVED** [review]
Good catch

File: deploy/terraform/main.tf [hunk 1]
```diff
@@ -73,9 +73,9 @@ resource "null_resource" "tink_directory" {
 
   provisioner "remote-exec" {
     inline = [
-      "iptables -A FORWARD -i eth1 -o bond0 -j ACCEPT",
-      "iptables -A FORWARD -i bond0 -o eth1 -m state --state ESTABLISHED,RELATED -j ACCEPT",
-      "iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE",
+      "iptables -A FORWARD -i enp1s0f1 -o bond0 -j ACCEPT",
```
Comment (mmlb) [inline deploy/terraform/main.tf:76]: I believe the machine type is an input the user can change, which could end up with different nic names. We already do nic name detection in setup.sh which I believe tf ends up calling can we make use of that info instead?

Comment (nshalman) [reply deploy/terraform/main.tf:76]: What about just moving these commands into setup.sh entirely?

Comment (mmlb) [reply deploy/terraform/main.tf:76]: sgtm, don't know why it was done here like this.

Comment (nshalman) [reply deploy/terraform/main.tf:76]: Yeah, there's a spot in setup.sh where IP forwarding gets turned on. That's the right spot.

**Comment by nshalman** [comment]
Updated per feedback from @mmlb. Looking for further feedback if this new approach is good.

**Review by mmlb — APPROVED** [review]

File: setup.sh [hunk 2]
```diff
@@ -110,6 +110,17 @@ setup_networking() (
 	else
 		echo "$ERR tinkerbell network interface configuration failed"
 	fi
+
+	if [ -r .nat_interface ]; then
+		NAT_INTERFACE=$(cat .nat_interface)
+	fi
+	if [ -n "$NAT_INTERFACE" ] && ip addr show "$NAT_INTERFACE" &>/dev/null; then
+		# TODO(nshalman) the terraform code would just run these commands as-is once
```
Comment (mmlb) [inline setup.sh:118]: ubuntu has some sort of iptables-persist package iirc. Yep `iptables-persistent`

Comment (nshalman) [reply setup.sh:118]: Should I try to get that wired up, or land as is with the TODO?

Comment (mmlb) [reply setup.sh:118]: todo is fine imo, hasn't been an issue so far I think.

**Comment by nshalman** [comment]
Squashed and ready to land if everyone likes this.

**Comment by markyjackson-taulia** [comment]
I say it is ready to ship
/shipit

**Pull request merged**